src/main/java/cn/nukkit/entity/passive/EntityVillager.java

```diff
@@ -76,60 +98,54 @@ public class EntityVillager extends EntityCreature implements InventoryHolder, E
-                        .setMaxUses(16)
-                        .setPriceMultiplierA(0.05f)
-                        .setRewardExp((byte) 1)
-                        .setTier(2)
-                        .setTraderExp(2)
-                        .build())
-                .add(RecipeBuildUtils.of(input1, output1)
-                        .setMaxUses(16)
-                        .setPriceMultiplierA(0.05f)
-                        .setRewardExp((byte) 1)
-                        .setTier(3)
-                        .setTraderExp(2)
-                        .build());
+    {
         this.tierExpRequirement = new int[]{0, 10, 70, 150, 250};
     }
 
     public EntityVillager(FullChunk chunk, CompoundTag nbt) {
         super(chunk, nbt);
     }
 
+    @Override
+    public IBehaviorGroup requireBehaviorGroup() {
+        return new BehaviorGroup(
+                this.tickSpread,
+                Set.of(
+                        //用于刷新InLove状态的核心行为
+                        new Behavior(
+                                new InLoveExecutor(400),
+                                all(
+                                        new PassByTimeEvaluator(CoreMemoryTypes.LAST_BE_FEED_TIME, 0, 400),
+                                        new PassByTimeEvaluator(CoreMemoryTypes.LAST_IN_LOVE_TIME, 6000, Integer.MAX_VALUE)
+                                ),
+                                1, 1
+                        ),
+                        //生长
+                        new Behavior(
+                                new AnimalGrowExecutor(),
+                                //todo：Growth rate
+                                all(
+                                        new PassByTimeEvaluator(CoreMemoryTypes.ENTITY_SPAWN_TIME, 20 * 60 * 20, Integer.MAX_VALUE),
+                                        entity -> entity instanceof EntityAnimal animal && animal.isBaby()
+                                )
+                                , 1, 1, 1200
+                        )
+                ),
+                Set.of(
+                        new Behavior(new FlatRandomRoamExecutor(0.4f, 12, 40, true, 100, true, 10), new PassByTimeEvaluator(CoreMemoryTypes.LAST_BE_ATTACKED_TIME, 0, 100), 4, 1),
+                        new Behavior(new EntityBreedingExecutor<>(EntityPig.class, 16, 100, 0.5f), entity -> entity.getMemoryStorage().get(CoreMemoryTypes.IS_IN_LOVE), 3, 1),
+                        new Behavior(new MoveToTargetExecutor(CoreMemoryTypes.NEAREST_FEEDING_PLAYER, 0.4f, true), new MemoryCheckNotEmptyEvaluator(CoreMemoryTypes.NEAREST_FEEDING_PLAYER), 2, 1),
+                        new Behavior(new LookAtTargetExecutor(CoreMemoryTypes.NEAREST_PLAYER, 100), new ProbabilityEvaluator(4, 10), 1, 1, 100),
+                        new Behavior(new FlatRandomRoamExecutor(0.2f, 12, 100, false, -1, true, 10), (entity -> true), 1, 1)
+                ),
+                Set.of(new NearestFeedingPlayerSensor(8, 0), new NearestPlayerSensor(8, 0, 20)),
```

Comment:
If we use NearestFeedingPlayerSensor, the Entity must extends EntityAnimal. However it's not the case for villager now.